database/migrations/2020_03_26_203841_create_attendance_exports_table.php

```diff
@@ -0,0 +1,39 @@
+<?php
+
+declare(strict_types=1);
+
+use Illuminate\Database\Migrations\Migration;
+use Illuminate\Database\Schema\Blueprint;
+use Illuminate\Support\Facades\Schema;
+
+class CreateAttendanceExportsTable extends Migration
+{
+    /**
+     * Run the migrations.
+     */
+    public function up(): void
+    {
+        Schema::create('attendance_exports', static function (Blueprint $table): void {
+            $table->increments('id');
+            $table->string('secret')->unique();
+            // useCurrent() for all of these because the migration required a default value.
+            $table->timestamp('start_time')->useCurrent();
+            $table->timestamp('end_time')->useCurrent();
+            $table->timestamp('expires_at')->useCurrent();
```

Comment:
Seems like these should be nullable instead?

Reply:
I don't think they should be nullable as null values don't make sense for them. These defaults don't either but if you end up with them then the export can't export anything, which is a safe "failure".